intel-sgx/dcap-artifact-retrieval/src/provisioning_client/mod.rs

```diff
@@ -552,8 +554,22 @@ pub trait ProvisioningClient {
             pck_id.pce_isvsvn,
             Some(&pck_id.qe_id),
         )?;
-
-        pck_cert
+        let fmspc = pck_cert.sgx_extension()?.fmspc;
+        let tcb_info = self.tcbinfo(&fmspc)?;
+        let tcb_data = tcb_info.data()?;
+        let mut pcks = Vec::new();
+        for tcb_level in tcb_data.tcb_levels() {
+            let cpu_svn = tcb_level.tcb.cpu_svn();
+            let pce_isvsvn = tcb_level.tcb.pce_svn();
+            pcks.push(self.pckcert(
```

Comment:
For multipackage platforms, there may not be a pckcert for every tcb level. An error in getting a single `pckcert` should not result in an error for the overall get all pckcerts.

Reply:
That's true with Intel PCS. However, with PCCS if there are any PCKs cached for the platform, PCCS will return a "best available" PCK cert. I don't know the behavior for Azure though. Is there a reference I could check?

Reply:
And what happens when there isn't such an "best available" PCK cert? It can't just make up a `PckCert`, so I guess it must through an error, but that's not what you want here.

Reply:
A "best available" PCK cert also implies that we need to check what the tcb level is for the pck you get back.